R/tribble.R

```diff
@@ -82,19 +122,7 @@ tribble <- function(...) {
     )
   }
 
-  frame_mat <- matrix(frame_rest, ncol = frame_ncol, byrow = TRUE)
-  frame_col <- lapply(seq_len(ncol(frame_mat)), function(i) {
-    col <- frame_mat[, i]
-    if (any(vapply(col, needs_list_col, logical(1L)))) {
-      col
-    } else {
-      unlist(col)
-    }
-  })
-
-  # Create a tbl_df and return it
-  names(frame_col) <- frame_names
-  as_tibble(frame_col)
+  list(frame_names = frame_names, frame_rest = frame_rest)
```

Comment:
The error checking and the composition of the returned list could also form a separate function.